scripts/generate.py

```diff
@@ -819,8 +830,10 @@ def generate_initializer(self, file_out):
         '''
         # __init__ keyword arguments
         kwargs = []
-        for key, content in chain(self.inherited_optional.items(),
-                                  self.optional_user_defined.items()):
+        for key, content in chain(
+                self.optional_user_defined.items(),
+                self.inherited_optional.items(),
+        ):
```

Comment:
I see that this change in order has resulted in the changes in the generated __init__ signatures.  Is this intentional?
Just an observation: I suppose we cannot fix the signature such that future additional optional arguments would only be added at the end of a signature.  Somewhere (in the doc maybe) the users should be advised to use named arguments when instantiating meta classes.